### PR TITLE
 refactor(ir): rename `.output_dtype` and `.output_shape` to `.dtype` and `.shape` respectively  

### PR DESCRIPTION
**Pull request by kszucs**
Depends on #5899 

prefer shorter names for these attributes, aliases are provided for backwards compatibility but they are deprecated

### CODE REVIEW

**Comment by kszucs** [comment]
@cpcloud rebased it after #5899 has been merged

File: ibis/common/tests/test_patterns.py [hunk 1]
```diff
@@ -255,21 +255,21 @@ class Value(Generic[T, S], Coercible):
         def __coerce__(cls, value, T=..., S=...):
             return cls(value, Scalar())
 
-        def output_dtype(self) -> T:
+        def dtype(self) -> T:
             ...
 
-        def output_shape(self) -> S:
```
Comment (cpcloud) [inline ibis/common/tests/test_patterns.py:261]: Should we keep these around?

Comment (cpcloud) [reply ibis/common/tests/test_patterns.py:261]: Sorry, not in the test case, but in the implementation.

**Review by cpcloud — APPROVED** [review]
LGTM, much nicer!

File: ibis/common/tests/test_patterns.py (hunk 1 above)
Comment (cpcloud) [reply ibis/common/tests/test_patterns.py:261]: Ah, you're already doing it. Great!

File: ibis/common/typing.py [hunk 2]
```diff
@@ -66,7 +66,10 @@ class properties.
             if isinstance(attr, property):
                 annots = _get_type_hints(attr.fget, include_extras=include_extras)
                 if return_annot := annots.get("return"):
+                    # pick the first return annotation if there are multiple
+                    # if name not in hints:
                     hints[name] = return_annot
+                    # hints.setdefault(name, return_annot)
```
[the remark this answers is not in the document]
Comment (kszucs) [reply ibis/common/typing.py:72]: Yes, removing.

**Pull request merged**